### PR TITLE
SonarQube 7.3

### LINKED ISSUE #11: SonarQube 7.3
- https://www.sonarqube.org/sonarqube-7-3
- https://jira.sonarsource.com/jira/secure/ReleaseNote.jspa?projectId=10930&version=14464

### PR DESCRIPTION
**Pull request by mwz**
With this PR our master branch will become where we develop against the latest SonarQube and the existing `6.x` series (for SonarQube 6.7 LTS) will be maintained on a separate branch.

~This is not yet ready to be merged, I'm just opening the PR for visibility.~

Closes #11 and #7.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -36,7 +36,7 @@ sourceGenerators in Compile ++= Seq(
 )
 
 // Lib dependencies
-val sonarVersion = "6.7"
+val sonarVersion = "7.3"
 libraryDependencies ++= List(
```
Comment (BalmungSan) [inline build.sbt]: I remember, I once said that when we refactored the old scalastyle module, there will be some dependencies we could remove... I think it was `sonar-core` and `guava`, but I'm not sure. I will try to search for that discussion after the review. But if I couldn't find it, could you please try removing some dependencies and check if we don't need?

**Comment by BalmungSan** [comment]
Nice, I think we could remove [this file](https://github.com/mwz/sonar-scala/blob/sonar-7.3/src/test/scala/org/sonar/api/utils/log/SonarLogTester.scala) too? - or did I get wrong your previous explanation?
Also, I almost sure this will close #7 too, or ar we missing something? _(PS: I don't know if github allow to close multiple issues with one PR automatically, so we don't have to do it manually)_.

**Comment by mwz** [comment]
> Nice, I think we could remove [this file](https://github.com/mwz/sonar-scala/blob/sonar-7.3/src/test/scala/org/sonar/api/utils/log/SonarLogTester.scala) too? - or did I get wrong your previous explanation?

It is actually used by the `LogSpec` and I guess we could also use in other places for testing logging, so we should keep it.

> Also, I almost sure this will close #7 too, or ar we missing something? _(PS: I don't know if github allow to close multiple issues with one PR automatically, so we don't have to do it manually)_.

I'm not sure either, but I'll add reference in the description.

**Comment by mwz** [comment]
I've tested this on example projects and sonar-scala and everything works as expected - I think this PR is ready to be merged. I'll prepare release notes and update readme in a separate PR.

**Comment by BalmungSan** [comment]
> "Should we also bump the default version of Scala up to `2.12` as a part of this release?"

I think yes, we should.

**Comment by mwz** [comment]
> So, are you releasing after the checks complete? - or what else you want to do?

No, I think that's it for now - I'll go ahead and make the release shortly!

**Pull request merged**